core/src/main/cpp/io/BufferedInputStream.cpp

```diff
@@ -52,7 +52,7 @@ void BufferedInputStream::mark(size_t readlimit)
 
 void BufferedInputStream::reset()
 {
-    if (readlimit_ == -1)
+    if (readlimit_ == (size_t)-1)
```

Comment:
Use a C++ static_cast<> instead of a C-cast.

There's also a `readlimit_ = -1` statement lower down. Does that have the same type mismatch problem? Should `readlimit_` be a different type to explicitly support signed integers? I'm wondering now if the -1 is incompatible. @padolph any thoughts?

Reply:
So long as size_t is the appropriate type here I will go ahead and make the change to use static_cast, and agree that would be better.  The assignment of `(size_t) = (int)` doesn't provide the same warning to that of when you compare the two types, but it probably should be at least called out in code, and we should probably use another cast there as well.

Reply:
Alan and I decided it is best to use std::numeric_limits<size_t>::max(), which is effectively the same as casting -1. We have to keep the member type size_t because of the arg of the function inherited from InputStream.

Reply:
I'll use std::numeric_limits<size_t>::max() to replace the assignment and comparisson from '-1'.

Reply:
It is the same, but it is also a valid input to `::mark(size_t)` that should enable buffering, while -1 disables buffering. The parameter type needs to be `size_t` but the internal variable `readlimit_` doesn't.

Reply:
We decided on a different strategy and now are going to use checked math stuff